src/vmemcache_repl.c

```diff
@@ -250,14 +287,19 @@ repl_p_lru_insert(struct repl_p_head *head, void *element,
 static void
 repl_p_lru_use(struct repl_p_head *head, struct repl_p_entry **ptr_entry)
 {
-	ASSERTne(ptr_entry, NULL);
+	struct repl_p_entry *entry;
 
-	util_mutex_lock(&head->lock);
+	ASSERTne(ptr_entry, NULL);
 
-	if (*ptr_entry != NULL)
-		TAILQ_MOVE_TO_TAIL(&head->first, *ptr_entry, node);
+	entry = util_val_compare_and_swap64(ptr_entry, *ptr_entry, NULL);
```

Comment:
What's the point of compare&swap if you don't care about the previous value?

Reply:
I do care. 'entry' is the previous value.